tests/timeseries/test_periodic_feature_extractor.py

```diff
@@ -0,0 +1,224 @@
+    assert np.allclose(feature_extractor.avg(start, end), [1.5, 1.5])
+
+    async def _test_fun(data, period, start, end, expected, weights):
+        feature_extractor = await init_feature_extractor(
+                data, timedelta(seconds=period)
+        )
+        ret = feature_extractor.avg(
+            UNIX_EPOCH + timedelta(seconds=start),
+            UNIX_EPOCH + timedelta(seconds=end),
+            weights=weights,
+        )
+        print("{}, {}, {}, {}: {}".format(period, start, end, data[start:end], ret))
+        assert np.allclose(ret, expected)
+
+    async def test_09(period, start, end, expected, weights=None):
+        data: List[float] = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]  # type: ignore[no-redef]
+        await _test_fun(data, period, start, end, expected, weights)
+
+    async def test_011(period, start, end, expected, weights=None):
+        data: List[float] = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11]  # type: ignore[no-redef]
+        await _test_fun(data, period, start, end, expected, weights)
+
+    print()
+    # empty time period
+    with pytest.raises(ValueError):
+        await test_09(period=5, start=1, end=1, expected=[4711])
+
+    # 3)
+    # moving window not multiple of period
+    with pytest.raises(ValueError):
+        await test_09(period=3, start=1, end=1, expected=[4711])
+
+    # 6)
+    # time period in moving window
+    await test_09(period=2, start=0, end=1, expected=[5])
+    await test_09(period=2, start=0, end=2, expected=[5, 6])
+    await test_09(period=2, start=5, end=7, expected=[4, 5])
+    await test_09(period=2, start=8, end=10, expected=[5, 6])
+    await test_09(period=5, start=0, end=1, expected=[5])
+    await test_09(period=5, start=5, end=6, expected=[5])
+
+    # 4+5)
+    # time period outside of moving window in future
+    await test_09(period=5, start=10, end=11, expected=[5])
+    # time period outside of moving window in past
+    await test_09(period=5, start=-5, end=-4, expected=[5])
+
+    await test_09(period=5, start=0, end=2, expected=[5, 6])
+    await test_09(period=5, start=0, end=3, expected=[5, 6, 7])
+    await test_09(period=5, start=0, end=4, expected=[5, 6, 7, 8])
+    await test_09(period=5, start=1, end=5, expected=[6, 7, 8, 9])
+
+    # No full time period in moving window, expect to throw
+    await test_09(period=5, start=0, end=5, expected=[5, 6, 7, 8, 9])
+    await test_09(period=5, start=0, end=6, expected=[5]) # should throw
+    await test_09(period=5, start=0, end=7, expected=[5, 6]) #should throw
+    await test_09(period=5, start=0, end=8, expected=[5, 6, 7]) #should throw
+    await test_09(period=5, start=0, end=9, expected=[5, 6, 7, 8]) #should throw
+    # 7)
+    #await test_09(period=5, start=0, end=10, expected=[4711]) #fails, should throw
+    await test_09(period=5, start=0, end=11, expected=[5]) #should throw
+    await test_09(period=5, start=0, end=12, expected=[5, 6]) #should throw
```

Comment:
After adding the missing check, all of these are throwing now. You can put them in a
`with pytest.raises(Exception):` block for testing.